### PR TITLE
implement post event/join endpoind #13

### LINKED ISSUE #13: POST /events/:id/volunteers
## Things to be done
- Create POST /events/:id/volunteers
- Write Unit tests
- Write Documentation for the endpoint

NOTE: For Detailed info please check the endpoint requirements from this [page](https://sehebi.notion.site/Define-Event-Fund-Endpoints-7d0644dbe9174a41b3b44cf82476e60d)

### PR DESCRIPTION
**Pull request by sarahmood**
### The following were added
- event/:id/join route
- validation params
- integration test
- swagger docs 

***To be considered:****
Test needs more attention as I encountered errors with code 401.

### CODE REVIEW

File: src/controllers/auth.js [hunk 1]
```diff
@@ -94,7 +94,7 @@ async function signUp(req, res) {
         const token = jwt.sign(payload, process.env.SECRET_KEY, {
             expiresIn: FOURTEEN_DAYS_STRING,
         });
-
```
Comment (muhammed-shihebi) [inline src/controllers/auth.js:97]: Please delete unneeded console logs onece you are done with them.

File: src/app.js [hunk 2]
```diff
@@ -16,10 +19,37 @@ app.use(encryptCookieNodeMiddleware(process.env.SECRET_KEY));
 
 app.use(express.urlencoded({ extended: false }));
 app.use(express.json());
-
+const options = {
```
Comment (muhammed-shihebi) [inline src/app.js]: In the dev branch, these options are moved to variables.js 
While merging your branch, just make sure to remove them from here and get the options from the variables.js file

File: src/routes/events.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+const {
+    EVENT_VALIDATION_RULES,
+    handleValidation,
+} = require('../utility/validation');
+
+const router = express.Router();
+const eventController = require('../controllers/events');
+
+router.post(
+    '/:id/join',
```
Comment (muhammed-shihebi) [inline src/routes/events.js]: Join is a verb. Endpoints should not contain verbs. Now you are saying something like: "Post user as join." Better is to say "Post user as volunteer." 
So, I think this should be changed to volunteer.

File: src/controllers/events.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
+const EventModel = require('../models/event');
+const { User } = require('../models/user');
+
+async function joinedVoulnteers(req, res) {
+    try {
+        const event = await EventModel.findById(req.params.id);
+        if (!event) {
+            return res.status(404).json({ message: 'Event not found' });
+        }
+        await EventModel.findByIdAndUpdate(req.params.id, {
```
Comment (muhammed-shihebi) [inline src/controllers/events.js]: You are not checking first if the user is alreay in the array of confirmedVolunteers. This array should not contain any duplicates.

File: src/controllers/events.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+            $push: { confirmedVolunteers: req.user.id },
+        });
+        await User.findByIdAndUpdate(req.user.id, {
```
Comment (muhammed-shihebi) [inline src/controllers/events.js:174]: Same as above.

File: src/docs/event.yaml [hunk 6]
```diff
@@ -0,0 +1,168 @@
+tags:
```
Comment (muhammed-shihebi) [inline src/docs/event.yaml:1]: This file contains some minor indentation issues. These issues are fixed in this gist: https://gist.github.com/muhammed-shihebi/9d195c5b5a69e27c2f8ad87c6107f730
I left you some comments so you know where the issues were, the gist fixes these issues.

File: src/docs/event.yaml [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+            - Event
+        summary: Join user to an event
+        description: add user to an event as a confirmed Volunteer
+        operationId: Join user
+        consumes:
+            - application/json
+        produces:
+            - application/json
+        security:
+            - auth_cookie: []
```
Comment (muhammed-shihebi) [inline src/docs/event.yaml]: You are using an auth_cookie without define it first. This is also fixed in the gist.

File: src/docs/event.yaml [hunk 8]
```diff
@@ -0,0 +1,168 @@
...
+              schema:
+                  type: string
+                  example: 62e297b6eb2bce7177c6fefc
+            - in: body
+              name: body
+              description: Updated event object
+              required: true
+              schema:
+                  $ref: '#/components/schemas/Event'
+    responses:
```
Comment (muhammed-shihebi) [inline src/docs/event.yaml]: Pay attention to indentation!

File: src/docs/event.yaml [hunk 9]
```diff
@@ -0,0 +1,168 @@
...
+            content:
+                application/json:
+                    schema:
+                        $ref: '#/components/schemas/ValidationError'
+        500:
+            description: Internal server error
+
+components:
+    schemas:
+    Event:
```
Comment (muhammed-shihebi) [inline src/docs/event.yaml]: This schema should be deleted since you are not using it.

File: src/routes/__tests__/events.test.js [hunk 10]
```diff
@@ -0,0 +1,90 @@
+const request = require('supertest');
```
Comment (muhammed-shihebi) [inline src/routes/__tests__/events.test.js]: Your test is not working for several reasons that are hard to explain. For that, I created a gist that contains a working test for your endpoint. The rest of the tests you can do yourself :)
https://gist.github.com/muhammed-shihebi/3b4ab4d00c73f5c78cb76ee681259bff

File: src/controllers/auth.js [hunk 11]
```diff
@@ -94,7 +94,7 @@ async function signUp(req, res) {
...
+        console.log(token);
```
Comment (muhammed-shihebi) [inline src/controllers/auth.js]: Please delete unneeded console logs onece you are done with them.

**Review by muhammed-shihebi — CHANGES_REQUESTED** [review]
Other than the test file and swagger, your code is very good and clean. I have suggested some changes in your endpoint and added examples for swagger and the test. Hope it helps 😁

**Review by muhammed-shihebi — APPROVED** [review]

File: src/controllers/events.js [hunk 12]
```diff
@@ -0,0 +1,35 @@
+const EventModel = require('../models/event');
+const { User } = require('../models/user');
+
+async function joinedVoulnteers(req, res) {
+    try {
+        const event = await EventModel.findById(req.params.id);
+        if (!event) {
+            return res.status(404).json({ message: 'Event not found' });
+        }
+        const JoinedUser = await EventModel.findOne({
```
Comment (halitbatur) [inline src/controllers/events.js]: Variables need to be camelCased

**Review by halitbatur — APPROVED** [review]
Great work! just 1 change

**Pull request merged**